### PR TITLE
Add final details to Tingley

### PR DESCRIPTION
**Pull request by h-mayorquin**
This is the pull request to add some further requirements from David Tingley lateral septum paper before we are reading for conversion.
- [x] Add epochs to the converter.
- [x] Create the parallel file.
- [x] Write explicitly the sessions that should be included from David Tingley email. 
- [ ] Generate the nwb files 
- [ ] Upload nwb files to dandi.
- [ ] Save the environment that can be used to reproduce the flow.
- [ ] Erase the nwb files that will not be used in the other Tingley conversion.

As things stand, the code should be ready for conversion.

### CODE REVIEW

File: buzsaki_lab_to_nwb/tingley_code_septal/tingleyseptalbehaviorinterface.py [hunk 1]
```diff
@@ -163,3 +164,26 @@ def run_conversion(self, nwbfile: NWBFile, metadata: dict):
                     )
             [table.add_row(**row) for row in sorted(data, key=lambda x: x["start_time"])]
             processing_module.add(table)
+
+        # Add epochs
+        lfp_file_path = session_path / f"{session_path.name}.lfp"
+        raw_file_path = session_path / f"{session_id}.dat"
+        xml_file_path = session_path / f"{session_id}.xml"
+
+        if raw_file_path.is_file():
```
Comment (h-mayorquin) [inline buzsaki_lab_to_nwb/tingley_code_septal/tingleyseptalbehaviorinterface.py:173]: @CodyCBakerPhD  your comments in slack make me thing that maybe there is a more standard way of doing this?  I am not sure that we can rely (can we?) on the recorders already being instantiated from the other data interfaces (as they are in the same `run_conversion` method as here in the behavioral interface) so I just went an extracted the data directly from the source.

File: buzsaki_lab_to_nwb/tingley_code_septal/convert_tingley_septal.py [hunk 2]
```diff
@@ -1,66 +1,68 @@
...
 import warnings
 
 from scipy.io import loadmat
 from nwb_conversion_tools.utils.json_schema import load_dict_from_file
 from nwb_conversion_tools.utils.json_schema import dict_deep_update
 
 from buzsaki_lab_to_nwb import TingleySeptalNWBConverter
+from joblib import Parallel, delayed
 
+n_jobs = 20
```
Comment (CodyCBakerPhD) [inline buzsaki_lab_to_nwb/tingley_code_septal/convert_tingley_septal.py:12]: Note that DANDIHub can support many more jobs than this if you're on a larger instance; to make the conversion go as fast as possible on run, you can max this out as long as `n_jobs * buffer_gb < max_RAM` (`max_RAM` is a DANDIHub instance value paired with the maximum number of cores allocated)

File: buzsaki_lab_to_nwb/tingley_code_septal/convert_tingley_septal.py [hunk 3]
```diff
@@ -116,3 +118,11 @@
         conversion_options=conversion_options,
```
Comment (CodyCBakerPhD) [inline buzsaki_lab_to_nwb/tingley_code_septal/convert_tingley_septal.py:118]: `iterator_opts=dict(buffer_gb=x)` is another way to make the full conversions run faster, `x=1 GB` by default, but can safely go up to about `x=max_RAM / n_jobs * 0.8` or so.

File: buzsaki_lab_to_nwb/tingley_code_septal/tingleyseptalbehaviorinterface.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply buzsaki_lab_to_nwb/tingley_code_septal/tingleyseptalbehaviorinterface.py:173]: @h-mayorquin In a more general context, yeah, I would have suggested handling this at the `NWBConverter.__init__` class level, after the `super()` initializes all the objects. (For `RecordingExtractor` objects you can also set these things at the object level via these functions: https://github.com/SpikeInterface/spikeextractors/blob/master/spikeextractors/baseextractor.py#L363

However, I think the way you're doing it here is just fine and deals with the addition in a safe, reliable manner.

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin LGTM, I'm just going to double check the files on the Hub to make sure the end time makes sense then I'll merge and it's ready to run

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin Yep, took a look at various stub files on DANDIHub and they look great; thanks for all your hard work on this! Go ahead with full conversion and upload

**Pull request merged**